internal/api/handlers.go

```diff
@@ -68,7 +68,10 @@ func GetRecommendationSetList(c echo.Context) error {
 		}
 	}
 
-	queryParams := MapQueryParameters(c)
+	queryParams, err := MapQueryParameters(c)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, err)
```

Comment:
Unsure, if we should send this error as JSON, could you provide a sample?

Reply:
We are very perticular with respect to converting the given start/end date to Time format, and so we are also raising the error in backend about we are unable to convert it. However at the user end if the problem is in format of the date then we just process further, however the results will be incorrect? So for sample we can use any date which is not in YYYY-MM-DD format and api works, which should not. With this change it shows error as below,
~~~
* Connection #0 to host localhost left intact
{
    "message": "parsing time \"1970-1-01\" as \"2006-01-02\": cannot parse \"1-01\" as \"01\"",
    "status": "error"
}
~~~

Reply:
We could send this error as json to user, however IMO we should use a standard date probably `1970-01-01` for comparison.

Reply:
Additionally, could we type assert the other params as well?

Reply:
> We could send this error as json to user, however IMO we should use a standard date probably `1970-01-01` for comparison in `start_date` and `end_date`.

We cant use 1970-01-01 as time.Parse function does not accept it as per RFC3339. So not doing this change

Reply:
> Additionally, could we type assert the other params as well?
> 
> Instead of implicit type casting i.e. using `:=` I think we should declare type explicitly and then grab the values using `c.QueryParams()`

Done!